src/main/java/com/github/splendor_mobile_game/database/InMemoryDatabase.java

```diff
@@ -109,6 +149,11 @@ public void loadCards() {
         }
     }
 
+    @Override
+    public ArrayList<Noble> getAllNobles(){
+        return this.allNobles;
```

Comment:
It'll better return a copy o this list. Returning a reference is prone to errors as we want `allNobles` list to be immutable.

Reply:
Then you can simplify the `getRandomNobles` method.

Reply:
But in getAllCards we also return reference